packages/ai-ollama/src/node/ollama-language-model.ts

```diff
@@ -37,30 +37,41 @@ export class OllamaModel implements LanguageModel {
     readonly providerId = 'ollama';
     readonly vendor: string = 'Ollama';
 
-    constructor(protected readonly model: string, protected host: () => string | undefined) {
-    }
-
-    get id(): string {
-        return this.providerId + '/' + this.model;
-    }
-
-    get name(): string {
-        return this.model;
+    /**
+     * @param id the unique id for this language model. It will be used to identify the model in the UI.
+     * @param model the unique model name as used in the Ollama environment.
+     * @param hostProvider a function to provide the host URL for the Ollama server.
+     * @param defaultRequestSettings optional default settings for requests made using this model.
+     */
+    constructor(
+        public readonly id: string,
+        protected readonly model: string,
+        protected host: () => string | undefined,
+        public defaultRequestSettings?: { [key: string]: unknown }
+    ) { }
+
+    protected getSettings(request: LanguageModelRequest): Partial<ChatRequest> {
+        const settings = request.settings ?? this.defaultRequestSettings ?? {};
+        return {
+            options: settings as Partial<Options>
+        };
     }
 
     async request(request: LanguageModelRequest, cancellationToken?: CancellationToken): Promise<LanguageModelResponse> {
+        const settings = this.getSettings(request);
+        console.log(JSON.stringify(settings));
         const ollama = this.initializeOllama();
 
         if (request.response_format?.type === 'json_schema') {
             return this.handleStructuredOutputRequest(ollama, request);
         }
         const response = await ollama.chat({
-            ...this.DEFAULT_REQUEST_SETTINGS,
             model: this.model,
+            ...this.DEFAULT_REQUEST_SETTINGS,
+            ...settings,
```

Comment:
This is now a shallow merge of options and therefore behaves differently than the other language models, e.g. huggingface. I think we should either always shallow merge or always exclusively use the handed over ones or default ones.

Reply:
@sdirix 
Ollama setting are inside the `options` property and the default don't provide them.

Reply:
@dhuebner I agree @sdirix If you also agree, please resolve the conversation.

Reply:
You're right, I overlooked that part. Minor: I would like to suggest to improve the naming here and for example set `options` here directly instead of hiding that behind a similarly named `settings`.